week6/hun-geun/newsapi/main.js

```diff
@@ -0,0 +1,92 @@
+/*API KEY*/
+const keyValue = '2c6a16d84bbb4f1387f78194e6f68f67'
+
+/*Event Listen*/
+const inputWord = document.querySelector('input');
+inputWord.addEventListener('keypress', (e) => {
+    if (e.key === "Enter") {
+        search();
+        inputWord.blur();
+    }
+})
+const searchButton = document.querySelector('button');
+searchButton.addEventListener('click', search)
+const ulList = document.querySelector('ul');
+
+function search(){
+    const keyWord = inputWord.value.trimStart()
+    keyWord ? getNewsData(keyWord) : alert("검색어를 입력하세요.")
+    inputWord.value = '';
+    const resultKey = document.querySelector('.result-key')
+    resultKey.innerHTML = `Result about '${keyWord}'`;
+}
+
+/* data 가져오기 */
+async function getNewsData(keyWord) {
+    const word = keyWord;
+    const ori = await fetch(`https://newsapi.org/v2/everything?q=${word}&sortBy=popularity&apiKey=${keyValue}`)
+    if (ori.status !== 200) {
+        throw new Error("Can't Search News");
+    } 
+    const newsData = await ori.json();
+    if (newsData.articles.length === 0){
+        alert(`${word}에 대한 검색결과가 없습니다.`)
+    } else {
+        dataProcess(newsData.articles);
+    }
+}
+
+/* 항목별 분류 */
+function dataProcess(data) {
+    let authorList = [];
+    let titleList = [];
+    let urlList = [];
+    let urlToImageList = [];
+    let publishedAtList = [];
+
+    data.forEach(el =>{
+        authorList.push(el.author);
+        titleList.push(el.title);
+        urlList.push(el.url);
+        urlToImageList.push(el.urlToImage);
+        publishedAtList.push(el.publishedAt.slice(0,10));
+    });
+    render(authorList, titleList, urlList, urlToImageList, publishedAtList);
+}
+
+/*html에 뿌리기*/
+function render(authorList, titleList, urlList, urlToImageList, publishedAtList) {
+    let repeat;
+    ulList.innerHTML='';
+    /*Error prevention*/
+    const resultNumber = document.querySelector('.result-number');
+    try {
+        const showResult = document.querySelector('.result');
+        showResult.className = 'result-show';
+    } catch{}
+
+    authorList.length <= 20 ? repeat = authorList.length : repeat = 20;
```

Comment:
재할당이 일어난다기 보다는 조건에 따라 다른 값을 할당하는 구문이어서 `repeat` 을 let 변수가 아닌 const로 선언 후 삼항연산자 사용하시는 게 어떨까 싶습니다.
```suggestion
    const reapeat = authorList.length <= 20 ? authorList.length : 20;
```